lib/ts/utils.ts

```diff
@@ -205,3 +206,29 @@ export function validateTheStructureOfUserInput(
         });
     }
 }
+
+export async function getAPIVersionFromFileIfExists(): Promise<string | null> {
+    try {
+        let apiVersionFromFile = await new Promise<Buffer>((resolve, reject) => {
+            readFile(API_VERSION_FILE_PATH, (err, data) => {
+                if (err !== undefined && err !== null) {
+                    reject(err);
+                }
+                resolve(data);
+            });
+        });
+        return apiVersionFromFile.toString();
+    } catch (err) {
+        return null;
+    }
+}
+
+export async function storeAPIVersionInFile(apiVersion: string) {
+    try {
+        await new Promise((resolve, reject) => {
+            writeFile(API_VERSION_FILE_PATH, apiVersion, {}, () => {
```

Comment:
- If the files does not exist, does it create a new one?
- Does this overwrite the file if it already exists?
- We should also check that `/temp` exists before attempting to write to it, to be sure that we do not implicitly create one.
- Abstract away reading and writing to files within `/temp` folder into it's own function that does the above check as well.

Reply:
> If the files does not exist, does it create a new one?

No, but the logic has now been changed

> Does this overwrite the file if it already exists?

yes

> We should also check that /temp exists before attempting to write to it, to be sure that we do not implicitly create one.

yes, the directory will now be created if not already exists

> Abstract away reading and writing to files within /temp folder into it's own function that does the above check as well.

done